crates/core/src/common.rs

```diff
@@ -44,6 +45,56 @@ pub async fn get_remote_agents_near_location(
     filter_local_agents(local_agent_store, near_location).await
 }
 
+/// Get all responsive remote agents from the peer store.
+///
+/// Uses the [DynPeerStore] to get all agents, the [DynLocalAgentStore] to filter out local
+/// agents and the [DynPeerMetaStore] to filter out unresponsive agents.
+pub async fn get_all_responsive_remote_agents(
+    peer_store: DynPeerStore,
+    local_agent_store: DynLocalAgentStore,
+    peer_meta_store: DynPeerMetaStore,
+) -> K2Result<Vec<Arc<AgentInfoSigned>>> {
+    let all_agents = peer_store.get_all().await?;
+    let remote_agents =
+        filter_local_agents(local_agent_store, all_agents).await?;
+    filter_unresponsive_agents(peer_meta_store, remote_agents).await
+}
+
+/// Get responsive remote agents by overlapping storage arc.
+///
+/// Uses the [DynPeerStore] to get agents with overlapping storage arcs and the [DynLocalAgentStore]
+/// to filter out local agents.
```

Comment:
```suggestion
/// Uses the [DynPeerStore] to get agents with overlapping storage arcs, the [DynLocalAgentStore]
/// to filter out local agents and the [DynPeerMetaStore] to filter out unresponsive agents.
```

To match the function above.